task.go

```diff
@@ -192,7 +193,13 @@ func (e *Executor) RunTask(ctx context.Context, call taskfile.Call) error {
 		if err != nil {
 			return err
 		}
-		if upToDate {
+
+		preCondMet, err := e.areTaskPreconditionsMet(ctx, t)
```

Comment:
I think this should run before `upToDate, err := e.isTaskUpToDate(ctx, t)` because that function has side-effects when using the `checksum` method.